src/pudl/analysis/mcoe.py

```diff
@@ -147,14 +147,21 @@ def fuel_cost(pudl_out):
 
     # Split up the plants on the basis of how many different primary energy
     # sources the component generators have:
-    gen_w_ft = pd.merge(pudl_out.gen_eia923(),
+    gen_w_ft = pd.merge(pudl_out.gens_eia860()[['plant_id_eia',
+                                                'report_date',
+                                                'plant_name_eia',
+                                                'plant_id_pudl',
+                                                'generator_id',
+                                                'utility_id_eia',
+                                                'utility_name_eia',
+                                                'utility_id_pudl',
+                                                'fuel_type_count',
+                                                'fuel_type_code_pudl']],
                         pudl_out.hr_by_gen()[['plant_id_eia',
                                               'report_date',
                                               'generator_id',
-                                              'fuel_type_code_pudl',
-                                              'fuel_type_count',
                                               'heat_rate_mmbtu_mwh']],
-                        how='inner',
+                        how='outer',
```

Comment:
Did you really mean for this to be `outer`? I thought that was just for testing purposes. Is this to make sure that every generator ends up in the fuel cost outputs, even if there's no heat rate data associated with it?

Reply:
It was a test at first, but on further reflection I wanted to keep it this way. I am generally of the mind that it is okay (and possibly preferable) for these outputs to have all of the possible generators in them, with nulls for generators that don't have all of the parts. It makes it easier in future mergers and use to know it is always all there.